File: Snooper/src/main/java/com/prateekj/snooper/customviews/NonScrollListView.java
package com.prateekj.snooper.customviews;

import android.annotation.TargetApi;
import android.content.Context;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.ListView;

import static android.os.Build.VERSION_CODES.LOLLIPOP;

public class NonScrollListView extends ListView {
  public NonScrollListView(Context context) {
    super(context);
  }

  public NonScrollListView(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public NonScrollListView(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }

  @TargetApi(LOLLIPOP)
  public NonScrollListView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
    super(context, attrs, defStyleAttr, defStyleRes);
  }

  @Override
  protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
    int heightMeasureSpec_custom = MeasureSpec.makeMeasureSpec(
      Integer.MAX_VALUE >> 2, MeasureSpec.AT_MOST);
    super.onMeasure(widthMeasureSpec, heightMeasureSpec_custom);
    ViewGroup.LayoutParams params = getLayoutParams();
    params.height = getMeasuredHeight();
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/customviews/NonScrollListView.kt
package com.prateekj.snooper.customviews

import android.annotation.TargetApi
import android.content.Context
import android.os.Build.VERSION_CODES.LOLLIPOP
import android.util.AttributeSet
import android.widget.ListView

class NonScrollListView : ListView {
  constructor(context: Context) : super(context)

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(
    context,
    attrs,
    defStyleAttr
  )

  @TargetApi(LOLLIPOP)
  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(
    context,
    attrs,
    defStyleAttr,
    defStyleRes
  )

  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
    val heightMeasureSpecCustom = MeasureSpec.makeMeasureSpec(
      Integer.MAX_VALUE shr 2, MeasureSpec.AT_MOST
    )
    super.onMeasure(widthMeasureSpec, heightMeasureSpecCustom)
    val params = layoutParams
    params.height = measuredHeight
  }
}
